src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.Runtime.cs

```diff
@@ -57,6 +60,77 @@ public sealed override IEnumerable<RuntimeTypeHandle> TryGetImplementedInterface
             return RuntimeAugments.TryGetImplementedInterfaces(typeHandle);
         }
 
+        public sealed override void VerifyInterfaceIsImplemented(RuntimeTypeHandle typeHandle, RuntimeTypeHandle ifaceHandle)
+        {
+            if (RuntimeAugments.IsInterface(typeHandle))
+            {
+                throw new ArgumentException(SR.Argument_InterfaceMap);
+            }
+
+            if (!RuntimeAugments.IsInterface(ifaceHandle))
+            {
+                throw new ArgumentException(SR.Arg_MustBeInterface);
+            }
+
+            foreach (RuntimeTypeHandle iface in RuntimeAugments.TryGetImplementedInterfaces(typeHandle))
+            {
+                if (iface.Equals(ifaceHandle))
+                {
+                    return;
+                }
+            }
+
+            throw new ArgumentException(SR.Arg_NotFoundIFace);
+        }
+
+        public sealed override void GetInterfaceMap(Type instanceType, Type interfaceType, out MethodInfo[] interfaceMethods, out MethodInfo[] targetMethods)
+        {
+            AppContext.TryGetSwitch("Switch.System.Reflection.Type.GetInterfaceMap.DoNotThrowForIncompleteMap", out bool doNotThrow);
```

Comment:
I am not sure whether these app context switches are worth it. If somebody uses this method, they are heavy users of reflection and need to have close to everything opted in for reflection.

Reply:
The default behavior is designed for someone trying to use the API directly.
The switches are for the cases where 3rd party code uses this API.
While its possible that `TargetMethods` contains null elements in the CoreCLR implementation, i doubt any caller handles that so i built the second switch which just returns the interface `MethodInfo` which i hope is good enough for most cases.

Reply:
We add config switches only once we have a proof that they are really needed, e.g. there is a library that depends on it and it is not feasible to fix that library. (This is true for both CoreCLR and CoreRT.)